lib/src/components/NxDrawer/__tests__/NxDrawer.test.tsx

```diff
@@ -0,0 +1,283 @@
+    const drawer = getDrawer({ id: 'drawer-id', lang: 'en_US' });
+    expect(drawer).toHaveAttribute('id', 'drawer-id');
+    expect(drawer).toHaveAttribute('lang', 'en_US');
+  });
+
+  it('adds the nx-drawer--narrow class when the narrow variant prop is specified', function() {
+    const drawer = getDrawer({ variant: 'narrow' });
+    expect(drawer).toHaveClass('nx-drawer--narrow');
+  });
+
+  describe('NxDrawer event listener support', () => {
+    it('executes onClose callback when ESC key is pressed', function() {
+      const mockOnClose = jest.fn();
+
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      getDrawer({ onClose: mockOnClose })!;
+
+      const dialog = screen.getByRole('dialog', { hidden: true });
+      expect(mockOnClose).not.toHaveBeenCalled();
+
+      act(() => {
+        fireEvent.keyDown(dialog, { key: 'Escape' });
+      });
+
+      expect(mockOnClose).toHaveBeenCalled();
+    });
+
+    it('executes onClose ONLY when pressing ESC key', async function () {
+      const user = userEvent.setup();
+
+      const mockOnClose = jest.fn();
+
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      const drawer = getDrawer({ onClose: mockOnClose })!;
+
+      await user.keyboard('Tab');
+      await user.keyboard('Enter');
+      await user.keyboard('q');
+      await user.keyboard('Q');
+
+      fireEvent.animationEnd(drawer);
+
+      expect(mockOnClose).not.toHaveBeenCalled();
+    });
+
+    it('executes onCancel when drawer is closed and animation is completed', async function() {
+      const user = userEvent.setup();
+      const mockOnCancel = jest.fn();
+
+      const Fixture = () => {
+        const [open, toggleDrawer] = useToggle(true);
+        return (
+          <NxDrawer open={open} onClose={toggleDrawer} onCancel={mockOnCancel}>
+            <NxDrawer.Header>
+              <NxDrawer.HeaderTitle>Title</NxDrawer.HeaderTitle>
+            </NxDrawer.Header>
+          </NxDrawer>
+        );
+      };
+
+      render(<Fixture />);
+
+      const closeButton = screen.getByRole('button', { name: 'Close', hidden: true });
+      const dialog = screen.getByRole('dialog', { hidden: true });
+
+      await act(async () => {
+        await user.click(closeButton);
+        await fireEvent.animationEnd(dialog);
+      });
+
+      await waitFor(() => expect(mockOnCancel).toHaveBeenCalled());
+    });
+
+    it('executes onClose when clicked outside of the drawer', async function() {
+      const documentAddEventListener = document.addEventListener;
+
+      try {
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        const eventMap: any = {};
+
+        document.addEventListener = jest.fn((eventType: string, callback: () => void) => {
+          eventMap[eventType] = callback;
+        }) as jest.Mock;
+
+        const mockOnClose = jest.fn();
+
+        render(
+          <div>
+            <NxButton>Outside</NxButton>
```

Comment:
Let's make this outside element something non-interactive to ensure that clicking _anything_ outside of the drawer will close it.